### PR TITLE
chore: new user invitation independent files

### PR DESCRIPTION
**Pull request by Riddhiagrawal001**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [X] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

This PR contains all the files required for new user invitation page .

## How did you test it?
cannot test as it is a chore change

## Where to test it?

- [X] INTEG
- [ ] SANDBOX
- [ ] PROD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/UserManagement/UserRevamp/UserManagementHelper.res [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    let showToast = ToastState.useShowToast()
+    let internalSwitch = OMPSwitchHooks.useInternalSwitch()
+    let profileList = Recoil.useRecoilValueFromAtom(HyperswitchAtom.profileListAtom)
+    let {userInfo: {userEntity}} = React.useContext(UserInfoProvider.defaultContext)
+
+    let disableSelect = switch userEntity {
+    | #Organization | #Merchant => false
+    | #Profile => true
+    | _ => false
+    }
```
Comment (PritishBudhiraja) [inline src/screens/UserManagement/UserRevamp/UserManagementHelper.res:119]:
```suggestion
    let disableSelect = switch userEntity {
    | #Profile => true
    | #Organization | #Merchant
    | _ => false
    }
```

File: src/screens/UserManagement/UserRevamp/UserUtils.res [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  let response = groupsList->Array.reduce([], (acc, value) => {
+    if userRoleAccessValueList->Array.includes(value) && value->String.includes("view") {
+      let _ = [acc->Array.push("View")]
+    } else if userRoleAccessValueList->Array.includes(value) && value->String.includes("manage") {
+      let _ = acc->Array.push("Manage")
+    }
+    acc
+  })
+  response
+}
```
Comment (PritishBudhiraja) [inline src/screens/UserManagement/UserRevamp/UserUtils.res:101]:
```suggestion
let groupsAccessWrtToArray = (groupsList, userRoleAccessValueList) =>
  groupsList->Array.reduce([], (acc, value) => {
    if userRoleAccessValueList->Array.includes(value) && value->String.includes("view") {
      acc->Array.push("View")
    } else if userRoleAccessValueList->Array.includes(value) && value->String.includes("manage") {
      acc->Array.push("Manage")
    }
    acc
  })
```

File: src/screens/UserManagement/UserRevamp/UserUtils.res [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  (modulesWithAccess, modulesWithoutAccess)
+}
+
+let stringToVariantMapperForAccess = accessAvailable => {
+  open UserManagementTypes
+  switch accessAvailable {
+  | "View" => View
+  | "Manage" => Manage
+  | _ => View
+  }
```
Comment (PritishBudhiraja) [inline src/screens/UserManagement/UserRevamp/UserUtils.res:134]:
```suggestion
  switch accessAvailable {
  | "Manage" => Manage
  | "View"
  | _ =>
    View
  }
```

File: src/screens/UserManagement/UserRevamp/UserUtils.res [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+}
+
+let makeSelectBoxOptions = result => {
```
Comment (PritishBudhiraja) [inline src/screens/UserManagement/UserRevamp/UserUtils.res:137]: Where are you using this?

Comment (Riddhiagrawal001) [reply src/screens/UserManagement/UserRevamp/UserUtils.res:137]: Will be using in the next dependant PR

File: src/screens/UserManagement/UserRevamp/UserUtils.res [hunk 5]
```diff
@@ -0,0 +1,149 @@
+let moduleListRecoil: Recoil.recoilAtom<array<UserManagementTypes.userModuleType>> = Recoil.atom(
```
Comment (JeevaRamu0104) [inline src/screens/UserManagement/UserRevamp/UserUtils.res:1]: why atom is defined over here?

Comment (Riddhiagrawal001) [reply src/screens/UserManagement/UserRevamp/UserUtils.res:1]: As this atom is only used for the user management module

File: src/screens/UserManagement/UserRevamp/UserUtils.res [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+  }
+}
+
+let groupsAccessWrtToArray = (groupsList, userRoleAccessValueList) => {
+  groupsList->Array.reduce([], (acc, value) => {
+    if userRoleAccessValueList->Array.includes(value) && value->String.includes("view") {
+      let _ = [acc->Array.push("View")]
+    } else if userRoleAccessValueList->Array.includes(value) && value->String.includes("manage") {
+      let _ = acc->Array.push("Manage")
+    }
```
Comment (PritishBudhiraja) [inline src/screens/UserManagement/UserRevamp/UserUtils.res:97]:
```suggestion
    if userRoleAccessValueList->Array.includes(value) && value->String.includes("view") {
      acc->Array.push("View")
    } else if userRoleAccessValueList->Array.includes(value) && value->String.includes("manage") {
      acc->Array.push("Manage")
    }
```

**Review by PritishBudhiraja — DISMISSED** [review]

File: src/screens/UserManagement/UserRevamp/UserUtils.res (hunk 5 above)
Comment (Riddhiagrawal001) [reply src/screens/UserManagement/UserRevamp/UserUtils.res]: Moved the atom to HyperswitchAtom

**Review by PritishBudhiraja — DISMISSED** [review]

File: src/screens/UserManagement/UserRevamp/UserUtils.res [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+      } else {
+        listofemails->Array.forEach(ele => {
+          if ele->JSON.Decode.string->Option.getOr("")->HSwitchUtils.isValidEmail {
+            errors->Dict.set("email", "Please enter a valid email"->JSON.Encode.string)
+          }
+        })
+        ()
+      }
+    | String(roleType) =>
+      if roleType->String.length === 0 {
```
Comment (JeevaRamu0104) [inline src/screens/UserManagement/UserRevamp/UserUtils.res:67]: use logicUtils

File: src/screens/UserManagement/UserRevamp/UserUtils.res [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+  let errors = Dict.make()
+  open LogicUtils
+  let valuesDict = values->getDictFromJsonObject
+
+  fieldsToValidate->Array.forEach(key => {
+    let value = valuesDict->getJsonObjectFromDict(key)
+
+    switch value {
+    | Array(listofemails) =>
+      if listofemails->Array.length === 0 {
```
Comment (JeevaRamu0104) [inline src/screens/UserManagement/UserRevamp/UserUtils.res:56]: same here

Comment (Riddhiagrawal001) [reply src/screens/UserManagement/UserRevamp/UserUtils.res:41]: No LogicUtils present for empty array check

**Review by JeevaRamu0104 — APPROVED** [review]

**Review by PritishBudhiraja — APPROVED** [review]

**Pull request merged**